commercial-schedules/src/main/java/org/dcsa/conformance/standards/cs/action/CsAction.java

```diff
@@ -0,0 +1,31 @@
+package org.dcsa.conformance.standards.cs.action;
+
+import java.util.Map;
+import java.util.function.Supplier;
+import org.dcsa.conformance.core.scenario.ConformanceAction;
+import org.dcsa.conformance.standards.cs.party.SuppliedScenarioParameters;
+
+public abstract class CsAction extends ConformanceAction {
+
+  protected final Supplier<SuppliedScenarioParameters> sspSupplier;
+  protected final int expectedStatus;
+
+  public CsAction(
+      String sourcePartyName,
+      String targetPartyName,
+      ConformanceAction previousAction,
+      String actionTitle,
+      int expectedStatus) {
+    super(sourcePartyName, targetPartyName, previousAction, actionTitle);
+    this.sspSupplier = _getSspSupplier(previousAction);
+    this.expectedStatus = expectedStatus;
+  }
+
+  private Supplier<SuppliedScenarioParameters> _getSspSupplier(ConformanceAction previousAction) {
+    return previousAction instanceof SupplyScenarioParametersAction supplyAvailableTdrAction
+        ? supplyAvailableTdrAction::getSuppliedScenarioParameters
```

Comment:
Tdr refers to one of the terms in EBL, please provide other naming for this variable.